File: libraries/analytics/src/main/java/com/tokopedia/analytics/debugger/domain/GetFpmLogUseCase.java
package com.tokopedia.analytics.debugger.domain;

import com.tokopedia.abstraction.base.view.adapter.Visitable;
import com.tokopedia.analytics.debugger.data.repository.FpmLogLocalRepository;
import com.tokopedia.analytics.debugger.data.repository.FpmLogRepository;
import com.tokopedia.usecase.RequestParams;
import com.tokopedia.usecase.UseCase;

import java.util.List;

import javax.inject.Inject;

import rx.Observable;

/**
 * @author okasurya on 5/16/18.
 */
public class GetFpmLogUseCase extends UseCase<List<Visitable>> {
    private FpmLogRepository fpmLogRepository;

    @Inject
    GetFpmLogUseCase(FpmLogLocalRepository fpmLogRepository) {
        this.fpmLogRepository = fpmLogRepository;
    }

    @Override
    public Observable<List<Visitable>> createObservable(RequestParams requestParams) {
        return fpmLogRepository.get(requestParams);
    }
}

File: libraries/analytics_debugger/src/main/java/com/tokopedia/analyticsdebugger/debugger/domain/GetFpmLogUseCase.kt
package com.tokopedia.analyticsdebugger.debugger.domain

import com.tokopedia.abstraction.base.view.adapter.Visitable
import com.tokopedia.analyticsdebugger.debugger.data.repository.FpmLogLocalRepository
import com.tokopedia.analyticsdebugger.debugger.data.repository.FpmLogRepository
import com.tokopedia.usecase.RequestParams
import com.tokopedia.usecase.UseCase

import javax.inject.Inject

import rx.Observable

/**
 * @author okasurya on 5/16/18.
 */
class GetFpmLogUseCase @Inject
internal constructor(fpmLogRepository: FpmLogLocalRepository) : UseCase<List<Visitable<*>>>() {
    private val fpmLogRepository: FpmLogRepository

    init {
        this.fpmLogRepository = fpmLogRepository
    }

    override fun createObservable(requestParams: RequestParams): Observable<List<Visitable<*>>> {
        return fpmLogRepository.get(requestParams)
    }
}
